src/main/java/school/lemon/changerequest/java/io/pr1/task1/NumberSearchEngineImp.java

```diff
@@ -0,0 +1,40 @@
+package school.lemon.changerequest.java.io.pr1.task1;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by User on 03.02.2017.
+ */
+public class NumberSearchEngineImp implements NumberSearchEngine {
+
+
+    @Override
+    public boolean hasNumberWithIndex(String filename, int index) throws FileNotFoundException, IndexOutOfBoundsException {
+
+        if (filename.length() < index) throw new IndexOutOfBoundsException();
```

Comment:
Not sure why do you compare file name length with index?
As per javadocs: `throws IndexOutOfBoundsException if given index in negative or 0`